core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/BillingApiServiceV2Spec.scala

```diff
@@ -917,11 +920,6 @@ class BillingApiServiceV2Spec extends ApiServiceSpec with MockitoSugar {
       verify(services.samDAO).deleteUserPetServiceAccount(ArgumentMatchers.eq(project.googleProjectId),
                                                           any[RawlsRequestContext]
       )
-      verify(services.samDAO).deleteResource(
```

Comment:
Hot Take: These checks add no value, and we should remove them. The only thing they check is if we've set up enough mocks to not throw unexpected NPEs.

Reply:
Not sure I agree with that. These are verifying that the Sam call to delete the `billing-project` resource is actually being made by whatever service makes that call. Without it, we won't know if there was a bug in `unregisterBillingProject` and it wasn't actually trying to delete the `billing-project` resource which might lead to us leaking Sam resources.

Whether _this_ is the place for us to verify that is a different question. It doesn't really feel like the best place for it, but I'd prefer to see the check get moved to the proper place than get deleted entirely

Reply:
Those checks are already done in unit tests for the appropriate classes (and with considerably more specificity).
These were written when the overall unit test coverage for the project was much sparser.

For the record, I wasn't necessarily saying they needed to be removed _now_ - more just provoking discussion (and a bit of venting frustration, tbh).